designs/package-spec.md

```diff
@@ -0,0 +1,98 @@
+# Package Controller Configuration
+
+## Introduction
+
+**Problem:** currently the ability to configure the package controller is very limited and requires code changes
+
+The preference in the CLI is to validate the values passed to the package controller, so the EKS Anywhere cluster specification will include all the necessary configuration values for the controller.
+
+### Tenets
+
+* ***Validation:*** allow the customer the flexibility to configure the package controller and validate those values.
+
+### Goals and Objectives
+
+As an EKS-A cluster administrator I want to configure the package controller for:
+
+* Allow packages only to be configured from a private registry
+* Harbor using proxy cache
+* Harbor using replication
+* Harbor as a private registry using the import images command
+* Configure ECR token refresher
+* Configure for kubelet private registry authentication
+* Support proxy configuration
+* Support package controller installation from alternate registries
+* Disable installation of package controller
+
+### Statement of Scope
+
+**In scope**
+* All of the above
+
+## Current state
+The package controller can be used currently with:
+1. Proxy configurations
+1. ECR token refresher enabled/disabled
+1. Installation of the package controller from alternate registries
+
+## Overview of Solution
+Add a section to the cluster specification to enable/disable the package controller installation and pass configuration values.
+
+## Solution Details
+
+A sample package controller section in the cluster specification. The values specified here:
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Cluster
+metadata:
+  name: billy
+spec:
+  packageController:
+    disable: false
+    # -- defaultRegistry for all package helm charts.
+    defaultRegistry: public.ecr.aws/eks-anywhere
+    # -- defaultImageRegistry for all package images.
+    defaultImageRegistry: 783794618700.dkr.ecr.us-west-2.amazonaws.com
```

Comment:
Just for my understanding, what would be the difference between using these two and using the cluster top level registry mirror?

Reply:
This would override the settings of the top level registry mirror if the user wanted to pull packages from a different mirror or *only* wanted packages from a mirror.

Reply:
Ahh I see. I thought @vincentni's most recent changes to the top level registry mirror already allowed to do that (having separate registry mirrors for packages/eks-a or only one for packages).

Would this replace that functionality? Or am I missing something and it does something different?

Reply:
I'm not sure what change you are talking about, but as far as I know nothing else allows separate registries for EKS-A and packages now. Vincent has/had a PR out to have different registries for management vs. workload clusters recently. That is different.

Reply:
PR #4128 allows the user to map our registry to their registry, but this allows the user to specify a registry mirror for packages when they are NOT using a registry mirror for EKS-A

Reply:
okay, vincent tried it out and he was able to use that feature to replicate this behavior I'm looking for

Reply:
okay, removed the default registry settings